File: Toy-Language-Interpreter/DataTypes/MyIList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_Interpreter.DataTypes
{
    interface MyIList<T>
    {
        void Enqueue(T e);
        T Dequeue();
        bool IsEmpty();
        string ToString();
    }
}

File: Toy-Language-Interpreter/src/DataTypes/MyIList.java
package DataTypes;

import Exception.EmptyMyListException;

import java.util.Iterator;

/**
 * Created by vladc on 22.10.2016.
 */
public interface MyIList<T> {
    void enqueue(T e);
    T dequeue() throws EmptyMyListException ;
    boolean isEmpty();
    Iterator iterator();
}
